### PR TITLE
Documentation

### PR DESCRIPTION
**Pull request by tomkins**
It's not ready, but I'm creating this pull request to keep it open so we're actually moving forward.

### CODE REVIEW

File: docs/source/layouts.rst [hunk 1]
```diff
@@ -3,4 +3,143 @@ Layouts
...
+
+    @templates.attach(Post)
+    class NewsPost(PageLayout):
+        content = columns.Column(width=640)
+
+
+Template name
+=============
+
+By default the template name used for rendering the layout is based on the name
```
Comment (tomkins) [inline docs/source/layouts.rst]: No, it's actually just lower case.

File: docs/source/blocks.rst [hunk 2]
```diff
@@ -0,0 +1,146 @@
...
+
+.. code:: python
+
+    from glitter.blockadmin import blocks
+
+    from .models import HeadlineText
+
+
+    @blocks.register(HeadlineText, category='Common')
+    class HeadlineTextAdmin(blocks.BlockAdmin):
```
Comment (zatan) [inline docs/source/blocks.rst]: Is this should be just `HeadlineText` ? 
Naming is not very consistent in the documentation https://github.com/blancltd/django-glitter/pull/50/files#diff-1e2bb462d1d00960bfcc1d917a5f45eeR75

Comment (tomkins) [inline docs/source/blocks.rst]: Wouldn't that conflict with the model class?

Comment (zatan) [inline docs/source/blocks.rst]: Did we previously decided to name this as `HeadlineTextBlock` ?

Comment (tomkins) [inline docs/source/blocks.rst]: Don't think so.

Certain **models** are named `FooBlock`, and others are named `Foo`. I think we possibly wanted all models to be `FooBlock` in the past, but it was a bit annoying having to change the verbose_name every time.

However, now we've got blocks.py - maybe we should go through it again.

File: docs/source/layouts.rst [hunk 3]
```diff
@@ -3,4 +3,139 @@ Layouts
...
+
+
+    class Default(PageLayout):
+        content = columns.Column(width=640)
+
+
+Attributes
+----------
+
+:attr:`width`
```
Comment (zatan) [inline docs/source/layouts.rst]: This rendering weird can we just have


**_width_**
         
     The width of the column in pixels. lore ipsum

File: tox.ini [hunk 4]
```diff
@@ -31,6 +31,20 @@ basepython = python3.4
...
+    -rdocs/requirements.txt
+    restructuredtext_lint>=0.14,<0.15
+    doc8>=0.7.0,<0.8.0
+commands =
+    rst-lint README.rst
+    doc8 docs/source
+    sphinx-build -W -a -b html docs/source docs/build
+
 [flake8]
 max-line-length = 100
```
Comment (zatan) [inline tox.ini]: This should be changed to 99.

File: docs/source/installation.rst [hunk 5]
```diff
@@ -4,76 +4,70 @@ Installation
...
+        # Glitter pages dependencies
+        'mptt',
+        'django_mptt_admin',
+        'sorl.thumbnail',
 
-.. note::
-    
-    Explain diffences between ``glitter`` and ``glitter.pages``
+        #...
+    )
```
Comment (zatan) [inline docs/source/installation.rst]: We have to mention that the user has to run `manage.py migrate` after adding required packages into `INSTALLED_APPS`

File: docs/source/layouts.rst (hunk 3 above)
Comment (tomkins) [inline docs/source/layouts.rst]: Updated, will ponder reusing it when it can actually link to something at a later date.

File: tox.ini (hunk 4 above)
Comment (tomkins) [inline tox.ini]: Would prefer to not touch code with this branch, you'll be bringing this in with add-tags anyway.

File: docs/source/installation.rst [hunk 6]
```diff
@@ -4,76 +4,70 @@ Installation
...
+urls.py
+=======
 
-The middleware ``glitter.middleware.ExceptionMiddleware`` is handles              
-exceptions if the object doesn't have the current version or hasn't been
-published it prompts the user to create a new page it also deals with blocks
-which raise exception ``GlitterRedirectException``::
+The Glitter block admin needs registering in your ``urls.py``::
 
+    from glitter.blockadmin import blocks
```
Comment (zatan) [inline docs/source/installation.rst]: This is wrong it should be `from glitter import block_admin`

File: docs/source/installation.rst [hunk 7]
```diff
@@ -4,76 +4,70 @@ Installation
...
 
-    MIDDLEWARE_CLASSES = [
+    urlpatterns = [
         #...
-        'glitter.pages.middleware.PageFallbackMiddleware',
-        'glitter.middleware.ExceptionMiddleware',
+        url(r'^blockadmin/', include(blocks.site.urls)),
```
Comment (zatan) [inline docs/source/installation.rst]: Needs replacing with `url(r'^blockadmin/', block_admin.site.urls),`

**Comment by zatan** [comment]
We missed the bit how to add the `Page` through the admin and start playing with the page and blocks straight away.

File: docs/source/installation.rst (hunk 5 above)
Comment (tomkins) [inline docs/source/installation.rst]: Updated.

File: docs/source/installation.rst (hunk 7 above)
Comment (tomkins) [inline docs/source/installation.rst]: Nope!

For the last released version, you would.

However the work you've done has moved this to `glitter.blockadmin.blocks`, so it's `blocks.sites.urls`.

Comment (zatan) [inline docs/source/installation.rst]: My bad! :dagger:

**Comment by tomkins** [comment]
I think we need lots more documentation - more developer documentation, a better quick start guide (~5 commands if possible), and user documentation. However - I'd rather get this branch merged instead of waiting for perfection.

Is this sufficient for now? We can always add more later, ideally as separate tickets/issues.

**Comment by zatan** [comment]
Yeah, sure I think I mentioned before I want to merge this ASAP as it getting out of date and no use for anyone!

**Pull request merged**